image/image.go

```diff
@@ -456,6 +470,13 @@ func bootstrapToRootDir(tsto *ToolingStore, model *asserts.Model, opts *Options,
 		if info.Base != "" && !local.hasName(snaps, info.Base) {
 			return fmt.Errorf("cannot add snap %q without also adding its base %q explicitly", name, info.Base)
 		}
+		// warn about missing default providers
+		dps := neededDefaultProviders(info)
+		for _, dp := range dps {
```

Comment:
`dps` does not appear to be used anywhere else, so you could pull it in the range loop
```
for _, dp := range neededDefaultProviders(info) {
    ...
}
```